.github/workflows/deploy.yml

```
@@ -1,68 +1,209 @@
+  cluster_name: webservers
+
+  ecr_repository_rails: "webserver-rails"
+  dockerfile_rails: "docker/rails.Dockerfile"
+
+  ecr_repository_nginx: "webserver-nginx"
+  dockerfile_nginx: "docker/nginx.Dockerfile"
+
+  ecr_repository_anycable_go: "webserver-anycable-go"
+  dockerfile_anycable_go: "docker/anycable-go.Dockerfile"
+
+  # TODO: Move this to be a secret
+  s3_bucket: "exercism-assets-staging"
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    name: Upload images to ECR, assets to S3, and defintion to ECS
+    runs-on: ubuntu-latest
+    steps:
+      ###############
+      ###############
+      # Setup Steps #
+      ###############
+      ###############
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: satackey/action-docker-layer-caching@v0.0.5
+        continue-on-error: true
+        with:
+          # Change these to rest teh cache
+          key: docker-layer-caching-deploy-{hash}
+          restore-keys: docker-layer-caching-deploy-
+          concurrency: 30
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_DEPLOY_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_DEPLOY_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.aws_region }}
+
+      - name: Login to Amazon ECR
+        id: login_to_ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      #########################
+      #########################
+      # Build the Rails image #
+      #########################
+      #########################
+      - name: Build and Push Rails Docker
+        env:
+          ECR_REGISTRY: ${{ steps.login_to_ecr.outputs.registry }}
+          ECR_REPOSITORY: ${{ env.ecr_repository_rails }}
+          IMAGE_TAG: ${{ github.sha }}
+          DOCKERFILE: ${{ env.dockerfile_rails }}
+          IMAGE_NAME: "${{ steps.login_to_ecr.outputs.registry}}/${{ env.ecr_repository_rails }}:${{ github.sha }}"
+        run: |
+          # Build a docker container and push it to ECR
+          docker build -t $IMAGE_NAME -f $DOCKERFILE .
+          docker push $IMAGE_NAME
+
+          # Retag this as the latest
+          docker tag $IMAGE_NAME $ECR_REGISTRY/$ECR_REPOSITORY:latest
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:latest
+
+          # Retag it as website below because the `:` hurts us otherwise
+          docker tag $IMAGE_NAME website
+
+          # Save the output for writing to the task definiton
+          echo "::set-output name=image::$IMAGE_NAME"
+
+      ##################################
+      ##################################
+      # Extact and upload assets to s3 #
+      ##################################
+      ##################################
+      - name: Copy assets from Dockerfile
+        run: |
+          container_id=$(docker create website)
+          docker cp ${container_id}:/opt/exercism/website/current/public/packs packs
```

_(@neenjaw This was that bit of bash you helped me with 🙂)_

---

This is a well-known pattern to get data out of Docker container. Could you also remove the container at the end?

```suggestion
          docker cp ${container_id}:/opt/exercism/website/current/public/packs packs
          docker rm ${container_id}
```

---

Added. Thanks 🙂